### PR TITLE
#87470 Show Process Version on Progress Tab

### PR DESCRIPTION
**Pull request by Klaas-Ritense**
Screenshot 2024-01-02 at 13 30 17

### CODE REVIEW

File: projects/valtimo/dossier/src/lib/components/dossier-detail/tab/progress/progress.component.css [hunk 1]
```diff
@@ -18,3 +18,14 @@
   height: calc(100vh - 380px);
   width: 100%;
 }
+
+.process-info-row {
+  display: flex;
+  flex-direction: row;
```
Comment (teodor-ritense) [inline projects/valtimo/dossier/src/lib/components/dossier-detail/tab/progress/progress.component.css]: flex-direction is row by default, no need to specify it

File: projects/valtimo/dossier/src/lib/components/dossier-detail/tab/progress/progress.component.html [hunk 2]
```diff
@@ -14,31 +14,52 @@
...
-    <label><strong>Process</strong></label
-    ><br />
-    <select class="form-control" (change)="loadProcessInstance($event.target.value)">
-      <option
-        *ngFor="let processDocumentInstance of processDocumentInstances"
-        [value]="processDocumentInstance.id.processInstanceId"
-        [selected]="selectedProcessInstanceId === processDocumentInstance.id.processInstanceId"
+  <div *ngIf="!obs.processInstanceItems">
+    <span>{{ 'progress.noProcessDocumentInstances' | translate }}</span>
+  </div>
```
Comment (teodor-ritense) [inline projects/valtimo/dossier/src/lib/components/dossier-detail/tab/progress/progress.component.html]:
```suggestion
    <span *ngIf="!obs.processInstanceItems">{{ 'progress.noProcessDocumentInstances' | translate }}</span>
```

File: projects/valtimo/dossier/src/lib/components/dossier-detail/tab/progress/progress.component.html [hunk 3]
```diff
@@ -14,31 +14,52 @@
...
+  <div class="process-info-row">
+    <div *ngIf="obs.processInstanceItems">
```
Comment (teodor-ritense) [inline projects/valtimo/dossier/src/lib/components/dossier-detail/tab/progress/progress.component.html]: Add the ngIf directly on the combo-box and remove the div

File: projects/valtimo/dossier/src/lib/components/dossier-detail/tab/progress/progress.component.ts [hunk 4]
```diff
@@ -14,43 +14,75 @@
...
 })
-export class DossierDetailTabProgressComponent implements OnInit {
-  public processDocumentInstances: ProcessDocumentInstance[];
-  public selectedProcessInstanceId: string;
-  public readonly documentId: string;
+export class DossierDetailTabProgressComponent {
+  private readonly processDocumentInstances$: Observable<Array<ProcessDocumentInstance>> = of(
+    this.route.snapshot.paramMap.get('documentId')
+  ).pipe(
+    switchMap(documentId => this.documentService.findProcessDocumentInstances(documentId)),
```
Comment (teodor-ritense) [inline projects/valtimo/dossier/src/lib/components/dossier-detail/tab/progress/progress.component.ts]: this.route.paramMap is an observable by default, no need to make it on your own from the snapshot. Check dossier-detail.component.ts on the canAssign$ attribute

File: projects/valtimo/dossier/src/lib/components/dossier-detail/tab/progress/progress.component.ts [hunk 5]
```diff
@@ -14,43 +14,75 @@
...
+    map(processDocumentInstances =>
+      processDocumentInstances.map(processDocumentInstance => {
+        processDocumentInstance.startedOn = new Date(processDocumentInstance.startedOn);
+        return processDocumentInstance;
+      })
```
Comment (teodor-ritense) [inline projects/valtimo/dossier/src/lib/components/dossier-detail/tab/progress/progress.component.ts]:
```suggestion
      processDocumentInstances.map(processDocumentInstance => ({
        ...processDocumentInstance,
        startedOn: new Date(processDocumentInstance.startedOn);
      }))
```

File: projects/valtimo/dossier/src/lib/components/dossier-detail/tab/progress/progress.component.ts (hunk 4 above)
Comment (Klaas-Ritense) [reply projects/valtimo/dossier/src/lib/components/dossier-detail/tab/progress/progress.component.ts]: Changed

File: projects/valtimo/dossier/src/lib/components/dossier-detail/tab/progress/progress.component.html (hunk 3 above)
Comment (Klaas-Ritense) [reply projects/valtimo/dossier/src/lib/components/dossier-detail/tab/progress/progress.component.html]: Changed

File: projects/valtimo/dossier/src/lib/components/dossier-detail/tab/progress/progress.component.css (hunk 1 above)
Comment (Klaas-Ritense) [reply projects/valtimo/dossier/src/lib/components/dossier-detail/tab/progress/progress.component.css]: Changed

**Review by teodor-ritense — APPROVED** [review]

**Pull request merged**